homeassistant/components/bluetooth_le_tracker/device_tracker.py

```diff
@@ -26,6 +27,15 @@ def setup_scanner(hass, config, see, discovery_info=None):
     # pylint: disable=import-error
     import pygatt
     new_devices = {}
+    adapter = None
+
+    async def async_stop():
+        nonlocal adapter
```

Comment:
It is `discover_ble_devices()` (where `adapter` is written to) that I thought should have `nonlocal` added.

Try adding a logging statement along with the kill, I would be surprised if it is executed.

Reply:
`nonlocal` needs to be set in the place that assigns a value to `adapter`

Reply:
sorted in the latest push. sorry, doubly stupid mistake.